### ISSUE TITLE
Contrast Limited AHE (CLAHE)

### ISSUE THREAD

**Issue by isears** [tensorflow/addons#2359]
TFA already includes a global histogram equalization function (`tfa.image.equalize`). I'm proposing adding a python implementation of CLAHE along side that.

**Relevant information**
- Are you willing to contribute it (yes/no): Yes
- Are you willing to maintain it going forward? (yes/no): Yes
- Is there a relevant academic paper? (if so, where): Yes
    - [S. M. Pizer, E. P. Amburn, J. D. Austin, et al.: Adaptive Histogram Equalization and Its Variations. Computer Vision, Graphics, and Image Processing 39 (1987) 355-368.](https://www.cs.unc.edu/Research/Image/MIDAG/pubs/papers/Adaptive%20Histogram%20Equalization%20and%20Its%20Variations.pdf)
- Is there already an implementation in another framework? (if so, where): Yes
    - [OpenCV](https://docs.opencv.org/master/d5/daf/tutorial_py_histogram_equalization.html)
    - [ImageMagick](https://imagemagick.org/script/clahe.php)
- Was it part of tf.contrib? (if so, where): No

**Which API type would this fall under (layer, metric, optimizer, etc.)**
image

**Who will benefit with this feature?**
This specific feature was requested a few years ago in tensorflow main (https://github.com/tensorflow/tensorflow/issues/24181), but it looks like no one has taken it on yet. It's especially useful for enhancing features on some medical imaging data.

**Any other info.**
Originally asked for guidance on the tensorflow developers mailing list and was directed to open a PR here. I have a draft already and it looks like it will fit into tensorflow addons pretty well.

**Comment by bhack** [comment]
CLAHE or porting MCLAHE to TF2? See https://github.com/tensorflow/tensorflow/issues/24181#issuecomment-760990668

**Comment by isears** [comment]
I'm proposing just normal CLAHE for grayscale or RGB images. The draft I have anticipates input shapes like `([batch_size], num_rows, num_columns, num_channels)`

**Comment by bhack** [comment]
Ok we put this in the ecosystem review pipeline

**Comment by tomerk** [comment]
Notes from ecosystem review: We think tfa.image is a good home for this

**Comment by bhack** [comment]
You can prepare a PR

**Comment by isears** [comment]
Awesome! I'm going to spend a few days cleaning up my draft. I'll plan on opening a PR sometime next week.

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
